### PR TITLE
Remove Dead MockTransport Code

### PR DESCRIPTION
**Pull request by original-brownbear**
* All these methods and `NoopGatewayAllocator` are unused

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ public TransportService createCapturingTransportService(Settings settings, Threa
                                                             @Nullable ClusterSettings clusterSettings, Set<String> taskHeaders) {
         StubbableConnectionManager connectionManager = new StubbableConnectionManager(new ConnectionManager(settings, this, threadPool),
             settings, this, threadPool);
-        connectionManager.setDefaultNodeConnectedBehavior((cm, discoveryNode) -> nodeConnected(discoveryNode));
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java]: This is a port of some changes from the `zen2` branch for easier future merging. Please do not back this out.

File: test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java [hunk 2]
```diff
@@ -253,10 +253,6 @@ protected void onSendRequest(long requestId, String action, TransportRequest req
         capturedRequests.add(new CapturedRequest(node, requestId, action, request));
     }
 
-    protected boolean nodeConnected(DiscoveryNode discoveryNode) {
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java]: This is a port of some changes from the `zen2` branch for easier future merging. Please do not back this out.

**Review by DaveCTurner — COMMENTED** [review]
Please don't make these changes to `CapturingTransport` - we have WIP using this extension point at the moment.

**Comment by original-brownbear** [comment]
@DaveCTurner reverted the chagnes to `CapturingTransport` :)

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/default-distro

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Comment by original-brownbear** [comment]
@DaveCTurner thanks!

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Pull request merged**